stonesoup/simulator/transition.py

```diff
@@ -0,0 +1,266 @@
+# -*- coding: utf-8 -*-
+from datetime import timedelta
+from copy import deepcopy
+
+import numpy as np
+
+from stonesoup.models.transition.linear import ConstantTurn, ConstantVelocity,\
+    CombinedLinearGaussianTransitionModel
+from stonesoup.types.array import StateVector
+
+
+def get_smooth_transition_models(initial_state, x_coords, y_coords, times, turn_rate):
+    """Generate a list of constant-turn and constant acceleration transition models alongside a
+    list of transition times to provide smooth transitions between 2D cartesian coordinates and
+    time pairs.
+
+    Parameters
+    ----------
+    initial_state: :class:`~.State` The initial state of the platform.
+    x_coords: A list of int/float x-coordinates (cartesian) in the order that the target must
+        follow.
+    y_coords: A list of int/float y-coordinates (cartesian) in the order that the target must
+        follow.
+    times: A list of :class:`~.datetime.datetime` dictating the times at which the target
+        must be at each corresponding coordinate.
+    turn_rate: Float
+        Angular turn rate (radians/second) measured anti-clockwise from positive x-axis.
+
+    Returns
+    -------
+    transition_models: A list of :class:`~.ConstantTurn` and
+        :class:`~.Point2PointConstantAcceleration` transition models.
+    transition_times: A list of :class:`~.datetime.timedelta` dictating the transition time
+        for each corresponding transition model in transition_models.
```

Comment:
Format of these isn't quite right, so they don't render properly. e.g. for first parameter
```
    Parameters
    ----------
    initial_state: :class:`~.State`
        The initial state of the platform.
```